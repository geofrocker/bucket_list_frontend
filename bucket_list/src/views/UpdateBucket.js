import React, {Component} from 'react';
import Header from '../components/Header';
import Sidebar from '../components/Sidebar';

class UpdateBucket extends Component{
    render(){
        return (
            <div>
                <Header/>
                <Sidebar/>
                <article className="content item-editor-page">
                    <div className="title-block">
                        <h3 className="title"> Update item
                            <span className="sparkline bar" data-type="bar"></span>
                        </h3>
                    </div>

                    <form name="item" method="post">
                        <div className="card card-block">
                            <div className="form-group row">
                                <label className="col-sm-2 form-control-label text-xs-right">
                                    Name:
                                </label>

                                <div className="col-sm-10">
                                    <input type="text"  id="bucket_name" className="form-control boxed" placeholder="" />
                                </div>
                            </div>
                            <div className="form-group row">
                                <label className="col-sm-2 form-control-label text-xs-right">
                                    Description:
                                </label>

                                <div className="col-sm-10">
                                    <input type="text" id="description" className="form-control boxed" placeholder="" />
                                </div>
                            </div>

                            <div className="form-group row">
                                <label className="col-sm-2 form-control-label text-xs-right">
                                    Category:
                                </label>
                                <div className="col-sm-10">
                                    <select className="c-select form-control boxed" id="category">
                                    </select>
                                </div>
                            </div>
                            <div className="col-sm-10 col-sm-offset-2">
                                <button type="submit" className="btn btn-primary">
                                    Update
                                </button>
                            </div>

                        </div>
                    </form>
                </article>
            </div>
        )
    }
}
export default UpdateBucket;