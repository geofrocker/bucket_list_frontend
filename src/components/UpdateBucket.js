import React, {Component} from 'react';
import Header from './Header';
import Sidebar from './Sidebar';
import {Redirect} from 'react-router-dom';
import axios from 'axios';
import swal from 'sweetalert';

class UpdateBucket extends Component{

    constructor(props){
        super(props);
        let bucket_id = this.props.match.params.bucket_id;
        this.state = {bucket_name: '', description: '', category: '', bucket_id:bucket_id, isAuthorized:false};
        this.handleUpdateName = this.handleUpdateName.bind(this);
        this.handleUpdateDescription = this.handleUpdateDescription.bind(this);
        this.handleUpdateCategory = this.handleUpdateCategory.bind(this);
        this.handleUpdate = this.handleUpdate.bind(this);
    }

    handleUpdateName(event){
        this.setState({bucket_name: event.target.value})
    }

    handleUpdateDescription(event){
        this.setState({description: event.target.value})

    }

    handleUpdateCategory(event){
        this.setState ({category: event.target.value})
    }

    componentDidMount() {
        axios({
            url: 'http://127.0.0.1:5000/api/v1/callback',
            method:"GET",
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            }
        }).then(()=>{
            this.setState({isAuthorized: true});
            window.localStorage.setItem('isLoggedIn', true)
        }).catch(()=>{
            this.setState({isAuthorized: false});
            window.localStorage.setItem('isLoggedIn', false)
        });

        axios({
            url: 'http://127.0.0.1:5000/api/v1/bucketlists/' + this.state.bucket_id,
            method: "GET",
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            }
        })
            .then((response)=>{
                let bucket_name = response.data.bucket.bucket_name;
                let description = response.data.bucket.description;
                let category = response.data.bucket.category;
                this.setState ({bucket_name: bucket_name, description:description, category:category })
            })
            .catch((xhr) =>{
                swal("Error", xhr.response.data.error, "error");
                if (xhr.response.status === 404){
                    this.setState({redirect:true})
                }

            });
    }

    handleUpdate(event){
        event.preventDefault();
        let data = {
            bucket_name:this.state.bucket_name,
            description:this.state.description,
            category:this.state.category
        };
        axios({
            url: 'http://127.0.0.1:5000/api/v1/bucketlists/' + this.state.bucket_id,
            method: 'PUT',
            data: data,
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            },
            datatype: "json"
        })
            .then(()=>{
                this.setState({redirect:true});
            })
            .catch((xhr) =>{
                swal("Error!", xhr.response.data.error, "error");

                if (xhr.response.status === 404){
                    this.setState({redirect:true})
                }

            });
    }

    render(){
        if(this.state.login_redirect){
            return(<Redirect to={'/login/'}/>)
        }

        if (this.state.redirect){
            return <Redirect to="/bucketlists/view"/>
        }

        if (!this.state.isAuthorized){
            return(
                <div>
                    <article className="content item-editor-page">

                    </article>
                </div>
            )
        }

        return (
            <div>
                <Header/>
                <Sidebar/>
                <article className="content item-editor-page">
                    <div className="title-block">
                        <h3 className="title"> Update item
                            <span className="sparkline bar"></span>
                        </h3>
                    </div>

                    <form method="post" id="update_bucket_form" onSubmit={this.handleUpdate}>
                        <div className="card card-block">
                            <div className="form-group row">
                                <label className="col-sm-2 form-control-label text-xs-right">
                                    Name:
                                </label>

                                <div className="col-sm-10">
                                    <input type="text" name="bucket_name" value={this.state.bucket_name} onChange={this.handleUpdateName} className="form-control boxed" placeholder="" />
                                </div>
                            </div>
                            <div className="form-group row">
                                <label className="col-sm-2 form-control-label text-xs-right">
                                    Description:
                                </label>

                                <div className="col-sm-10">
                                    <input type="text" name="description" value={this.state.description} onChange={this.handleUpdateDescription} className="form-control boxed" placeholder="" />
                                </div>
                            </div>

                            <div className="form-group row">
                                <label className="col-sm-2 form-control-label text-xs-right">
                                    Category:
                                </label>
                                <div className="col-sm-10">
                                    <input type="text" name="category" value={this.state.category} onChange={this.handleUpdateCategory} className="form-control boxed" placeholder="Eg Travel" />
                                </div>
                            </div>
                            <div className="col-sm-10 col-sm-offset-2">
                                <button type="submit" className="btn btn-primary">
                                    Update
                                </button>
                            </div>

                        </div>
                    </form>
                </article>
            </div>
        )
    }
}

export default UpdateBucket;