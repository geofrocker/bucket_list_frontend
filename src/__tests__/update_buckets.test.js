import React from 'react';
import UpdateBucket from '../components/UpdateBucket';

import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import {MemoryRouter} from 'react-router-dom';

describe('<UpdateBucket />', ()=>{
    beforeEach(() => {
        global.localStorage = {
            token: 'someToken',
            getItem: function () {
                return 'someToken'
            }
        };
    });

    it('contains  update bucket form', function () {
        const props = {
            match:{
                params: {bucket_id:1}
            }
        };
        const wrapper = mount(
            <MemoryRouter>
                <UpdateBucket {...props}/>
            </MemoryRouter>);

        const preventDefault = jest.fn();
        wrapper.setState({bucket_name:'test_bucket', description:"test description", isAuthorized:true});

        setTimeout(() => {
            expect(wrapper.find("#update_bucket_form").length === 1);
            wrapper.find("#update_bucket_form").simulate('submit', {preventDefault})

        }, 60);
        expect(toJson(wrapper)).toMatchSnapshot();


    });

    it('should change values in the form', function () {
        const props = {
            match:{
                params: {bucket_id:1}
            }
        };
        const wrapper = mount(
            <MemoryRouter>
                <UpdateBucket {...props}/>
            </MemoryRouter>);
        setTimeout(() => {
            wrapper.find("bucket_name").simulate('change', {target: {value: 'test bucket name'}});
            wrapper.find("description").simulate('change', {target: {value:'test description'}});
            wrapper.find("category").simulate('change', {target: {value:'test category'}});
        }, 60);
        expect(toJson(wrapper)).toMatchSnapshot();
    })



});

