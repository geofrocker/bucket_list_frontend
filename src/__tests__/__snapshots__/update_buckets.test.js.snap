// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateBucket /> contains  update bucket form 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "tghjog",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "tghjog",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <UpdateBucket
      match={
        Object {
          "params": Object {
            "bucket_id": 1,
          },
        }
      }
    >
      <div>
        <article
          className="content item-editor-page"
        />
      </div>
    </UpdateBucket>
  </Router>
</MemoryRouter>
`;

exports[`<UpdateBucket /> should change values in the form 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "gyfd60",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "gyfd60",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <UpdateBucket
      match={
        Object {
          "params": Object {
            "bucket_id": 1,
          },
        }
      }
    >
      <div>
        <article
          className="content item-editor-page"
        />
      </div>
    </UpdateBucket>
  </Router>
</MemoryRouter>
`;
